drizzlepac/hlautils/se_source_generation.py

```diff
@@ -387,46 +412,119 @@ def write_catalog(seg_cat, catalog_filename, product="tdp"):
+        seg_table["cxx"].description = "SExtractor Column cxx_image, ellipse parameter"
+        seg_table["cyy"].description = "SExtractor Column cyy_image, ellipse parameter"
+        seg_table["cxy"].description = "SExtractor Column cxy_image, ellipse parameter"
+        # FIX: is the mapping to _image or _world?
+        seg_table["covar_sigx2"].description = "SExtractor Column x2_image, (0,0) element of covariance matrix"
+        seg_table["covar_sigy2"].description = "SExtractor Column y2_image, (1,1) element of covariance matrix"
+        seg_table["covar_sigxy"].description = "SExtractor Column xy_image, (0,1) and (1,0) elements of covariance matrix"
+
+        seg_table["xmin"].description = "SExtractor Column xmin_image"
+        seg_table["xmax"].description = "SExtractor Column xmax_image"
+        seg_table["ymin"].description = "SExtractor Column ymin_image"
+        seg_table["ymin"].description = "SExtractor Column ymax_image"
+
+        # Write out the official filter detection product source catalog
+        seg_table["xcentroid"].info.format = ".10f"
+        seg_table["ycentroid"].info.format = ".10f"
         seg_table["RA_icrs"].info.format = ".10f"
         seg_table["Dec_icrs"].info.format = ".10f"
-        seg_table["RA_icrs"].unit = u.deg
-        seg_table["Dec_icrs"].unit =  u.deg
         print("seg_table (filter): {}".format(seg_table))
 
         seg_table.write(catalog_filename, format="ascii.ecsv")
         log.info("Wrote filter source catalog: {}".format(catalog_filename))
 
+def get_header_data(imghdu):
+
+    keyword_dict = {}
+
+    # Get the HSTWCS object from the primary header
+    keyword_dict["proposal_id"] = imghdu[0].header["PROPOSID"]
+    keyword_dict["image_file_name"] = imghdu[0].header['FILENAME']
+    keyword_dict["target_name"] = imghdu[0].header["TARGNAME"] 
+    keyword_dict["date_obs"] = imghdu[0].header["DATE-OBS"]
+    keyword_dict["instrument"] = imghdu[0].header["INSTRUME"]
+    keyword_dict["detector"] = imghdu[0].header["DETECTOR"]
+    keyword_dict["target_ra"] = imghdu[0].header["RA_TARG"]
+    keyword_dict["target_dec"] = imghdu[0].header["DEC_TARG"]
+    keyword_dict["expo_start"] = imghdu[0].header["EXPSTART"]
+    keyword_dict["expo_time"] = imghdu[0].header["EXPTIME"]
+    keyword_dict["ccd_gain"] = imghdu[0].header["CCDGAIN"]
+    keyword_dict["filter_tdp"] = imghdu[0].header["FILTER"]
+    keyword_dict["filter1"] = imghdu[0].header["FILTER1"]
+    keyword_dict["filter2"] = imghdu[0].header["FILTER2"]
+
+    # Get the HSTWCS object from the first extension
+    keyword_dict["wcs_name"] = imghdu[1].header["WCSNAME"]
+    keyword_dict["orientation"] = imghdu[1].header["ORIENTAT"]
+    keyword_dict["aperture_ra"] = imghdu[1].header["RA_APER"]
+    keyword_dict["aperture_dec"] = imghdu[1].header["DEC_APER"]
+    keyword_dict["aperture_pa"] = imghdu[1].header[" "]
+
+    return(keyword_dict)
+
+def annotate_table(input_table, keyword_dict, product="tdp"):
+
```

Comment:
If the metadata entry is used as the key for the dict, you should be able to apply those keywords in just a couple of lines with a simple loop over those keys, like "for key,value in dict.items(): if key not in key_list: input_table.meta[key] = value"
Some keys need additional processing, so "key_list" would be a list of keys that need extra work to get it into the metadata.